### PR TITLE
Fix choice loading from entities with identical property values

### PR DESCRIPTION
**Pull request by Romashka**
Fix choice loading from entities with identical property values

### CODE REVIEW

**Comment by greg0ire** [comment]
Your commit message body is good, but your commit message subject is not great. What if everybody uses "Update myFile.php" as a commit message? Not very meaningful, right? Please explain what you're doing in the commit message subject.

**Comment by greg0ire** [comment]
And please explain what bug you are "fixing". Telling what part of the project you are fixing is great, but telling what bug you are fixing is better? BTW, is it reported already?

**Comment by greg0ire** [comment]
Maybe you can show us the bug with a test?

**Comment by Romashka** [comment]
@greg0ire I've added test for this bug.

**Comment by greg0ire** [comment]
The test comes with the fix, there should be only one commit.
Without your fix, what happens?

File: Tests/Form/ChoiceList/ModelChoiceLoaderTest.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+        $modelChoiceLoader = new ModelChoiceLoader(
+            $this->modelManager,
+            'Sonata\AdminBundle\Tests\Fixtures\Entity\Foo',
+            'baz'
+        );
+
+        $expectedChoices = array(
+            $fooA->getBar() => sprintf('%s (id: %s)', $fooA->getBaz(), $fooA->getBar()),
+            $fooB->getBar() => sprintf('%s (id: %s)', $fooB->getBaz(), $fooB->getBar()),
```
Comment (greg0ire) [inline Tests/Form/ChoiceList/ModelChoiceLoaderTest.php]: I would refrain from calling getters and sprintf here. It's a test, harcoding the expected values would make it more readable and understandable IMO. Here it looks like you're writing a test for `Foo`

**Comment by Romashka** [comment]
@greg0ire If several entities have the same property value which should be displayed at choice list , the latter does not populates with such entities, only with the last one

**Comment by greg0ire** [comment]
Your commit message should read:

> What you're doing
>
> Why you're doing it

In other words :

> search value object in keys instead of values
>
> When the property chosen to generate choice list labels is not unique across entities, there should be one choice per entity anyway, and not one choice per label.

**Comment by greg0ire** [comment]
Don't hesitate to ask for help if you don't know how to fix up your commits together.

**Comment by greg0ire** [comment]
Looks better, but this should really be part of the commit message : 

> In the choices array, keys are values, whereas values are labels.
When the property chosen to generate choice list labels is not unique across entities, there should be one choice per entity anyway, and not one choice per label.

A commit message can be multiline, you know? Just use `git commit` instead of `git commit --message="your message"`

**Comment by greg0ire** [comment]
Good to merge IMO :+1: Thanks!

File: Tests/Form/ChoiceList/ModelChoiceLoaderTest.php (hunk 1 above)
Comment (OskarStark) [inline Tests/Form/ChoiceList/ModelChoiceLoaderTest.php]: after this change we can merge it

**Comment by Romashka** [comment]
@OskarStark the test has been fixed

**Comment by OskarStark** [comment]
Thank you @Romashka

**Pull request merged**